Barotrauma/BarotraumaShared/SharedSource/StatusEffects/StatusEffect.cs

```diff
@@ -1887,6 +1894,52 @@ protected void Apply(float deltaTime, Entity entity, IReadOnlyList<ISerializable
                 
                 TryTriggerAnimation(target, entity);
 
+                if (forceSayIdentifier != null) 
+                {
+                    Character targetCharacter = null;
+                    if (target is Character character)
+                    {
+                        targetCharacter = character;
+                    }
+                    string messagetosay = null;
+
+                    messagetosay = TextManager.Get(forceSayIdentifier).Value;
+
+                    if (messagetosay != null && targetCharacter != null && targetCharacter.SpeechImpediment < 100.0f && !targetCharacter.IsDead)
+                    {
+#if SERVER
+                        if (GameMain.Server != null)
+                        {
+                            ChatMessageType messagetype = ChatMessageType.Default;
+                            if (ChatMessage.CanUseRadio(targetCharacter, out WifiComponent radio) && forceSayInRadio) { messagetype = ChatMessageType.Radio; }
+                            GameMain.Server.SendChatMessage(messagetosay, messagetype, null, targetCharacter);
+                        }
+#endif
+#if CLIENT
+                        if (GameMain.GameSession?.CrewManager != null && GameMain.GameSession.CrewManager.IsSinglePlayer)
+                        {
+                            bool canUseRadio = ChatMessage.CanUseRadio(targetCharacter, out WifiComponent radio);
+                            ChatMessageType messagetype = ChatMessageType.Default;
+                            if (canUseRadio && forceSayInRadio)
+                            {
+                                messagetype = ChatMessageType.Radio;
+                            }
+                            string modifiedMessage = ChatMessage.ApplyDistanceEffect(messagetosay, messagetype, targetCharacter, Character.Controlled);
+                            if (!string.IsNullOrEmpty(modifiedMessage))
+                            {
+                                GameMain.GameSession.CrewManager.AddSinglePlayerChatMessage(targetCharacter.DisplayName, modifiedMessage, messagetype, targetCharacter);
+                            }
+                            if (canUseRadio && forceSayInRadio)
+                            {
+                                Signal s = new Signal(modifiedMessage, sender: targetCharacter, source: radio.Item);
+                                radio.TransmitSignal(s, sentFromChat: true);
+                            }
+                            targetCharacter.ShowSpeechBubble(ChatMessage.MessageColor[(int)messagetype], messagetosay);
+                        }
```

Comment:
[DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). There's quite a lot of code duplication here. I also feel like a lot of this logic is already defined elsewhere. That means if we need to do changes to the chat message logic, we need to do remember to do it in multiple places.